src/fides/api/ops/service/privacy_request/request_runner_service.py

```diff
@@ -684,15 +684,55 @@ def needs_batch_email_send(
     Delegates the "needs email" check to each configured email or
     generic email consent connector. Returns true if at least one
     connector needs to send an email.
+
+    If we don't need to send any emails, add skipped logs for any
+    relevant erasure and consent email connectors.
     """
+    can_skip_erasure_email: List[ConnectionConfig] = []
+    can_skip_consent_email: List[ConnectionConfig] = []
+
+    needs_email_send: bool = False
+
+    for connection_config in get_erasure_email_connection_configs(db):
+        if get_connector(connection_config).needs_email(  # type: ignore
+            user_identities, privacy_request
+        ):
+            needs_email_send = True
+        else:
+            can_skip_erasure_email.append(connection_config)
 
-    erasure_email_configs: Query = get_erasure_email_connection_configs(db)
-    consent_email_configs: Query = get_consent_email_connection_configs(db)
-    combined_configs = erasure_email_configs.union_all(consent_email_configs)
-    for connection_config in combined_configs:
+    for connection_config in get_consent_email_connection_configs(db):
         if get_connector(connection_config).needs_email(  # type: ignore
             user_identities, privacy_request
         ):
-            return True
+            needs_email_send = True
+        else:
+            can_skip_consent_email.append(connection_config)
+
+    if not needs_email_send:
+        _create_execution_logs_for_skipped_email_send(
+            db, privacy_request, can_skip_erasure_email, can_skip_consent_email
+        )
```

Comment:
In the request runner, if we determine there are no emails to send at all, but there are consent or erasure email connectors configured, add skipped logs for every email connector here.   Otherwise, we'll wait until the weekly email send to add the logs.